analyzers/src/SonarAnalyzer.CSharp/LiveVariableAnalysis/SonarCSharpLiveVariableAnalysis.cs

```diff
@@ -21,212 +21,211 @@
+    protected override IEnumerable<Block> Successors(Block block) =>
+        block.SuccessorBlocks;
 
-        protected override IEnumerable<Block> ReversedBlocks() =>
-            Cfg.Blocks.Reverse();
+    protected override IEnumerable<Block> Predecessors(Block block) =>
+        block.PredecessorBlocks;
+
+    protected override State ProcessBlock(Block block)
+    {
+        var ret = new SonarState(this, model);
+        ret.ProcessBlock(block);
+        return ret;
+    }
 
-        protected override IEnumerable<Block> Successors(Block block) =>
-            block.SuccessorBlocks;
+    private sealed class SonarState : State
+    {
+        private readonly SonarCSharpLiveVariableAnalysis owner;
+        private readonly SemanticModel model;
 
-        protected override IEnumerable<Block> Predecessors(Block block) =>
-            block.PredecessorBlocks;
+        public ISet<SyntaxNode> AssignmentLhs { get; } = new HashSet<SyntaxNode>();
 
-        protected override State ProcessBlock(Block block)
+        public SonarState(SonarCSharpLiveVariableAnalysis owner, SemanticModel model)
         {
-            var ret = new SonarState(this, semanticModel);
-            ret.ProcessBlock(block);
-            return ret;
+            this.owner = owner;
+            this.model = model;
         }
 
-        private sealed class SonarState : State
+        public void ProcessBlock(Block block)
         {
-            private readonly SonarCSharpLiveVariableAnalysis owner;
-            private readonly SemanticModel semanticModel;
-
-            public ISet<SyntaxNode> AssignmentLhs { get; } = new HashSet<SyntaxNode>();
-
-            public SonarState(SonarCSharpLiveVariableAnalysis owner, SemanticModel semanticModel)
+            foreach (var instruction in block.Instructions.Reverse())
             {
-                this.owner = owner;
-                this.semanticModel = semanticModel;
+                switch (instruction.Kind())
+                {
+                    case SyntaxKind.IdentifierName:
+                        ProcessIdentifier((IdentifierNameSyntax)instruction);
+                        break;
+
+                    case SyntaxKind.GenericName:
+                        ProcessGenericName((GenericNameSyntax)instruction);
+                        break;
+
+                    case SyntaxKind.SimpleAssignmentExpression:
+                        ProcessSimpleAssignment((AssignmentExpressionSyntax)instruction);
+                        break;
+
+                    case SyntaxKind.VariableDeclarator:
+                        ProcessVariableDeclarator((VariableDeclaratorSyntax)instruction);
+                        break;
+
+                    case SyntaxKind.AnonymousMethodExpression
+                    or SyntaxKind.ParenthesizedLambdaExpression
+                    or SyntaxKind.SimpleLambdaExpression
+                    or SyntaxKind.QueryExpression:
```

Comment:
The old `case` was better here due to alignment.

Reply:
Not a big fan of the extra `:` and `case`. Or reads more naturally. But not going to oppose to it.

Reply:
EDIT: In any case, I think I should not have touched this file.